lib/Runtime/Base/WindowsGlobalizationAdapter.cpp

```diff
@@ -829,20 +829,41 @@ if (this->object) \
         return ResolveLocaleLookup(scriptContext, locale, resolved);
     }
 
-    int IcuIntlAdapter::GetUserDefaultLocaleName(_Out_ LPWSTR lpLocaleName, _In_ int cchLocaleName)
+    size_t IcuIntlAdapter::GetUserDefaultLanguageTag(_Out_ char16* langtag, _In_ size_t cchLangtag)
     {
-        // XPLAT-TODO (doilij): implement GetUserDefaultLocaleName
-        const auto locale = _u("en-US");
-        const size_t len = 5;
-        if (lpLocaleName)
+        UErrorCode error = UErrorCode::U_ZERO_ERROR;
+        char bcp47[ULOC_FULLNAME_CAPACITY];
+        char defaultLocaleId[ULOC_FULLNAME_CAPACITY];
```

Comment:
I mused about this in comments in other places -- it's not too expensive to zero-out these arrays ` = { 0 }` and it's safer in case something goes wrong.
Now that we run this function only once on Intl initialization (thanks to caching the returned value in Intl.js), it seems worth it to pay that cost here.